src/Node/ClassStatementsGatherer.php

```diff
@@ -127,6 +138,10 @@ private function gatherNodes(Node $node, Scope $scope): void
 			}
 			return;
 		}
+		if ($node instanceof ClassPropertyNode && $scope->isInTrait()) {
+			$this->traitProperties[] = $node;
```

Comment:
Can you please explain what are we gaining by having this in a separate `traitProperties` property? Are there certain rules that will not see this and that only read `getProperties()`?

Reply:
I was kind of foreseeing problems with that in https://github.com/phpstan/phpstan-src/pull/1339. I was afraid that `UnusedPrivatePropertyRule` could start to get very annoying. E.g. for traits that define lots of private helper functions / props and my change would enforce classes to use all of them basically, right?

Reply:
Yeah, could you create a specific test case here in the UnusedPrivatePropertyRuleTest to show it's correct not to read trait properties? Then I'll merge it :)

Reply:
Are https://github.com/phpstan/phpstan-src/pull/1339/files#diff-d00cc4d3adb967370b742212a5a33d739fb44e1525953ac656f90c782f808b09 and tests/PHPStan/Rules/DeadCode/UnusedPrivatePropertyRuleTest.php enought? As you can see I had to adapt them already in that other PR 😅
Or were you thinking about something more specific?

Reply:
Oh sure, I haven't noticed, I thought you had to edit AnalyserIntegrationTest only. Ready to go then :)